analysis/figures/figure2.R

```diff
@@ -0,0 +1,166 @@
+  for (j in 1:length(prefixes)) {
+    factor_levels <- c(factor_levels, paste0(prefixes[j], outcomes[i]))
+  }
+}
+# Set factor levels 
+estimates_sub$grouping_name <- factor(estimates_sub$grouping_name, levels = factor_levels)
+
+
+# Set labels 
+labels <- c(
+  `All COVID-19-Acute_pancreatitis` = "All COVID-19
+  ",
+  `Hospitalised COVID-19-Acute_pancreatitis` = "Hospitalised COVID-19
+  Acute pancreatitis",
+  `Non-hospitalised COVID-19-Acute_pancreatitis` = "Non-hospitalised COVID-19
+  ",
+  `All COVID-19-Appendicitis` = "",
+  `Hospitalised COVID-19-Appendicitis` = "Appendicitis",
+  `Non-hospitalised COVID-19-Appendicitis` = "",
+
+  `All COVID-19-Gallstones_disease` = "",
+  `Hospitalised COVID-19-Gallstones_disease` = "Gallstones disease",
+  `Non-hospitalised COVID-19-Gallstones_disease` = "",
+  
+  `All COVID-19-Gastro_oesophageal_reflux_disease` = "",
+  `Hospitalised COVID-19-Gastro_oesophageal_reflux_disease` = "Gastro oesophageal reflux",
+  `Non-hospitalised COVID-19-Gastro_oesophageal_reflux_disease` = "",
+  
+  `All COVID-19-Ibs` = "",
+  `Hospitalised COVID-19-Ibs` = "Ibs",
+  `Non-hospitalised COVID-19-Ibs` = "",
+  
+  `All COVID-19-Lower_gi_bleeding` = "",
+  `Hospitalised COVID-19-Lower_gi_bleeding` = "Lower gi bleeding",
+  `Non-hospitalised COVID-19-Lower_gi_bleeding` = "",
+
+  `All COVID-19-Nonvariceal_gi_bleeding` = "",
+  `Hospitalised COVID-19-Nonvariceal_gi_bleeding` = "Nonvariceal gi bleeding",
+  `Non-hospitalised COVID-19-Nonvariceal_gi_bleeding` = "",
+  
+  `All COVID-19-Peptic_ulcer` = "",
+  `Hospitalised COVID-19-Peptic_ulcer` = "Peptic ulcer",
+  `Non-hospitalised COVID-19-Peptic_ulcer` = "",
+  
+  `All COVID-19-Upper_gi_bleeding` = "",
+  `Hospitalised COVID-19-Upper_gi_bleeding` = "Upper gi bleeding",
+  `Non-hospitalised COVID-19-Upper_gi_bleeding` = ""
+)
+
+
+# Function to plot 
+plot_estimates <- function(df, name) {
+  pd <- position_dodge(width = 0.5)
+
+  p <- ggplot(df, aes(x = outcome_time_median/7, y = hr, color = colour_cohort)) +
+    geom_line() +
+    geom_point(size = 2, position = pd) +
+    geom_hline(mapping = aes(yintercept = 1), colour = "#A9A9A9") +
+    geom_errorbar(
+      size = 1.2,
+      mapping = aes(
+        ymin = ifelse(conf_low < 0.25, 0.25, conf_low),
+        ymax = ifelse(conf_high > 64, 64, conf_high),
+        width = 0
+      ),
+      position = pd
+    ) +
+    scale_color_manual(values = levels(df$colour_cohort), labels = levels(df$cohort)) +
+    guides(color = guide_legend(nrow = 3)) +
+    guides(fill = ggplot2::guide_legend(ncol = 1, byrow = TRUE)) +
+    # facet_wrap(outcome ~ analysis, ncol = 3, scales = "free_x", strip.position = "top") +
+    theme_minimal() +
+    labs(x = "\nWeeks since COVID-19 diagnosis", y = "Hazard ratio and 95% confidence interval") +
+    scale_x_continuous(breaks = seq(0, max(df$outcome_time_median) / 7, 4)) +
+    scale_y_continuous(lim = c(0.25, 64), breaks = c(0.25, 0.5, 1, 2, 4, 8, 16, 64), trans = "log") +
```

Comment:
It might be useful to add 32 here, otherwise the figure 2 y-scale looks a bit odd.
```suggestion
    scale_y_continuous(lim = c(0.25, 64), breaks = c(0.25, 0.5, 1, 2, 4, 8, 16, 32, 64), trans = "log") +
```
However, we know that this part might be updated later on based on the results from Stata.